airbyte-ci/connectors/metadata_service/lib/metadata_service/models/src/AirbyteInternal.yaml

```diff
@@ -0,0 +1,23 @@
+---
+"$schema": http://json-schema.org/draft-07/schema#
+"$id": https://github.com/airbytehq/airbyte/airbyte-ci/connectors_ci/metadata_service/lib/models/src/AirbyteInternal.yml
+title: AirbyteInternal
+description: Fields for internal use only
+type: object
+additionalProperties: false
+properties:
```

Comment:
If they are required, we should add one to the invalid examples where the properties don't exist. 

I have the same overall question for `supportLevel` - is it a required property?

Reply:
Originally I had left it as not a required property so as not to break metadata processing at this time.

Once these are out and the systems are updated the plan is to cut a v2 of the metadata spec

[Deprecate releaseStage in Metadata Schema](https://app.zenhub.com/workspaces/connector-operations-6318ede5eda32e0010ef8566/issues/gh/airbytehq/airbyte/28440)

Reply:
I'm not sure how adding this as required would break metadata processing since you've manually patched all of the files so far. Can you elaborate?

Reply:
There are still old metadata files that can be processed.

For example at this time if you were to rerun the add metadata partition job in our orchestrator we would see a lot of failures for previous versions of metadata files.

This isnt a major issue, and overtime we expect there are old metadata files we cant/no longer want to process. Hell even right now we fail silently as long as this doesnt occur on a latest file path.

However I believe its good practise that while were still updating our downstream systems to use these new fields we allow them to exist harmoniously with the existing (old) systems. That way we can always rollback changes to downstream systems in case things go wrong.

Also it delays me from cutting the v2 of the metadata spec. And the more we delay their the more we learn. and the more we learn the less likely we end up in a situation where we have to cut a v3 immediately afterward.

Reply:
Ah true this is a good point, I always forget to think about old metadata files. 

Ok, then what I would recommmend is that we don't add the top level fields as required, but we do add the _ql and _sl fields as required and/or set default values, since they are nested! Anyone adding the new ab_internal tag should add those.

Reply:
Great idea! Also Ill add the defaults to the scaffold, but I want them to set them explicitly! If we add defaults here then they never have to add them to the metadata

Reply:
mm yeah i'm mostly concerned about community contributors - though I guess they'd not add the ab_internal stuff at all

Reply:
in that case no defaults but still reqired sub-keys makes sense to me!

Reply:
Fair call out. These are akward that way.

I think this means that the defaults for this fields are pushed to the consuming service.

In this case the transformation from metadata to registry entry

Reply:
Meaning for the Registry we can make these required and fill in the defaults when transforming from the metadata definition